### ISSUE TITLE
ListAppend no longer verifies

### ISSUE THREAD

**Issue by Sophietje** [utwente-fmt/vercors#431]
[ListAppend.java](https://github.com/utwente-fmt/vercors/blob/dev/examples/witnesses/ListAppend.java) no longer verifies even though it should pass (according to a comment in the file).

When I try to verify the file with silicon as a backend, it results in the following error:
```
[warning] (class vct.col.util.FeatureScanner): at file examples/witnesses/ListAppend.java from line 19 column 7 until line 115 column 1: detected inheritance
[warning] missing origin
Verification aborted exceptionally: java.util.concurrent.ExecutionException: java.util.NoSuchElementException: None.get
  Cause: java.util.NoSuchElementException: None.get
Errors! (4)
[warning] unspecified error exceptional.error
The final verdict is Fail
```

**Comment by bobismijnnaam** [comment]
This error happens because the resource is not final. What happens now is that the resource is encoded as an APF, which means the "list" predicate becomes an abstract predicate (this is a bug, because the class is marked as final. But apparently something changed since the last time I broke it). Since it is now an abstract predicate, you cannot unfold it, but the example does this (because it expects the predicate to be "normal"). VerCors should give an error in this case, not an exception. When I improve support for APFs this exception should disappear (and turn in a nice error). For now this issue can be fixed by adding "final" to the list resource (since that makes the type of the predicate statically known, which means unfolding works again).

What is more interesting is that this broke without travis complaining. Or did someone merge without waiting for travis, or did I oversee this exception the last time I broke it? (Since I modified inheritance & APF machinery a few weeks ago).
